Tweaks around website URL

elrido opened the pull request:
Changes:
- Ensure GitHub URL works when given an ssh URL or one ending in .git
- Ensure gpg agent is loaded, so password only needs to be typed in once
- Support for keyrings with multiple private keys (mine has several repo signing keys, several commit keys, several email keys, debian package signing key, rpm package signing key, ...) - this ensures the right one gets used, but requires the keys to be named in a certain way, so the script can remain generic
- Fix a minor typo

elrido committed:
Ensure GitHub URL works when given an ssh URL or one ending in .git

Signed-off-by: El RIDO <elrido@gmx.net>

elrido committed:
Typo

Signed-off-by: El RIDO <elrido@gmx.net>

elrido committed:
Ensure the relevant GPG release key is used

When having multiple GPG private keys on your keyring, this ensures the
one for the current project is picked. Requires that the key is stored
as "ProjectName release" in the key ring, where "ProjectName" is the
same as the last part of the GitHub URL, i.e.
git@github.com:OrganizationName/ProjectName

Signed-off-by: El RIDO <elrido@gmx.net>

rugk commented on signrelease.sh:
```diff
@@ -82,7 +85,7 @@ for ext in $ARCHIVE_TYPES; do
 
     # sign archive
-    gpg --armor --detach-sign "$TMP_DIR/${project}-${tag}.${ext}"
+    gpg -u "${originGitHub##*/} release" --armor --detach-sign "$TMP_DIR/${project}-${tag}.${ext}"
```
So this is the naming part?

Well… that is not much generic. It's just a coincidence that we chose this name for PrivateBin? 

Maybe better use that part up in the changable constants/variables at the top of the script? (If you want using a placeholder that gets replaced later like `GITHUB_REPO_NAME` or so?

elrido replied:
It's not a coincidence: I specifically labelled my key "PrivateBin release" and the above change codifies this. It is generic in that you can have a key per repo and it must be named "[Repo name] release". You can rename your local keys in the key ring to ensure none of your other keys get picked. I'm not aware of an easy way to have multiple key rings per user and tell gpg which ring to use...

rugk reviewed (changes requested):
Everything else LGTM.

elrido commented on signrelease.sh:
```diff
@@ -82,7 +85,7 @@ for ext in $ARCHIVE_TYPES; do
 
     # sign archive
-    gpg --armor --detach-sign "$TMP_DIR/${project}-${tag}.${ext}"
+    gpg -u "${originGitHub##*/} release" --armor --detach-sign "$TMP_DIR/${project}-${tag}.${ext}"
```
Thought about this one some more: How about we ask, like we do for other such options and just default it to the `"${project} relese"` form?

Would need to add this around lines 75 or 76:

```
read -rp "Enter the GPG private key name [${project} release]: " gpgPrivKey
gpgPrivKey=${gpgPrivKey:-$project release}
```

And then change line 88 to:

```suggestion
    gpg -u "${gpgPrivKey} release" --armor --detach-sign "$TMP_DIR/${project}-${tag}.${ext}"
```

Let me know if you'd be onboard with this change and I'll update the branch accordingly.

rugk replied:
Sounds like a good idea! (Nitpick: I would name the variable `gpgPrivateKey` as abrevations usually just reduce how understandable code is and I really like non-abbreviated variable names.)

BTW another idea is to get the key value (and maybe use that as a default) from the setting in `.gitconfig` (can be get via the `git config` command). Assuming this is the same key…

E.g.:
```
$ git config user.signingkey 
ABA9B8F6F448B07FD7EA4A1A05D40A636AFAB34D
```

As this is a common place to set such an option and one can set/overwrite such options in the usual git config hierarchy…

elrido replied:
I've update the branch with the question and your suggested term `gpgPrivateKey`. Using the git commit key would again not work for PrivateBin, as I use a different personal key for signing commits than the shared maintainer key for the releases. This may be extra effort, but at least the different identities can be separated.

elrido committed:
Ask user which GPG private key to use

In case multiple keys are present, for example a commit key and multiple
signing keys for different projects.
